assets/training/distillation/src/generate_data.py

```diff
@@ -389,11 +392,15 @@ def process_request(idx: str, data: dict, url: str, endpoint_key: str):
                         break
                     response_data = response.json()
                     # response content should be structured as below for a successful vllm response
-                    prediction_result = response_data["choices"][0]["message"][
-                        "content"
-                    ].strip()
+                    prediction_result = response_data["choices"][0]["message"]["content"].strip()
+
+                    # For CoT prompts, need to remove the reasoning and only use the answer
+                    if enable_cot and data_generation_task_type != DataGenerationTaskType.CONVERSATION:
```

Comment:
nit: Can we move this to a separate function which post process the response from the model. So that we can include post processing for other tasks without disturbing this function.

Reply:
Can add this change in a separate PR